test/e2e/self_hosted.go

```diff
@@ -48,6 +47,19 @@ type SelfHostedSpecInput struct {
 	SkipCleanup           bool
 	ControlPlaneWaiters   clusterctl.ControlPlaneWaiters
 	Flavor                string
+
+	// EnableUpgrade is used to test an upgrade of the self-hosted clusters kubernetes version.
+	EnableUpgrade bool
```

Comment:
TBD if we want this enabled or disabled per default

Enabling it per default makes it a lot more likely that providers are using the full test
WDYT about having this parameter similar to `SkipConformanceTests` in the Cluster upgrade test, aka `SkipUpgrade`?

I know that the used variables change, but I think that's fine given we can document this in v1.2-v1.3 md + usually folks have those set already for the upgrade test.

@chrischdi WDYT?

Reply:
I'm fine with that 👍  just went the other way around to keep less changes to the current state :-)